### ISSUE TITLE
Games do not run when launched with dedicated graphics card under X11 session

### ISSUE THREAD

**Issue by MrQvest** [Heroic-Games-Launcher/HeroicGamesLauncher#3247]
### Describe the bug

Games do not run with dedicated NVIDIA graphics card when under X11. 

Under Wayland all games launch as expected.

Games tested:

Bloons TD 6 (runs fine when on Wayland)
The Alto Collection (runs fine when on Wayland)
Darkwood (runs fine when on Wayland)

### Add logs

```text
(11:59:05) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(11:59:05) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(11:59:05) INFO:    [Connection]:       Connectivity: check-online
(11:59:05) INFO:    [Connection]:       Pinging external endpoints
(11:59:05) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2710.0"
  }
}
(11:59:06) WARNING: [Backend]:          Failed to register protocol with OS.
(11:59:06) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(11:59:06) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json --version
(11:59:06) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/nile_config /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/nile --version
(11:59:06) INFO:    [Connection]:       Connectivity: online
(11:59:06) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(11:59:06) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(11:59:06) INFO:    [Backend]:          
System Information:
CPU: 8x Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
Memory: 16.65 GB (used: 4.04 GB)
GPUs:
  GPU 0:
    Name: NVIDIA Corporation GP107M [GeForce GTX 1050 Ti Mobile]
    IDs: D=1c8c V=10de SD=118b SV=1025
    Driver: nvidia
  GPU 1:
    Name: Intel Corporation HD Graphics 630
    IDs: D=591b V=8086 SD=118a SV=1025
    Driver: i915
OS: Arch Linux 6.6.1-arch1-1 (linux)

The current system is not a Steam Deck
We are running inside a Flatpak container

Software Versions:
  Heroic: 2.10.0 Jewelry Bonney
  Legendary: 0.20.33 Undue Alarm
  gogdl: 0.7.3
  Nile: 1.0.0 Jonathan Joestar

(11:59:07) INFO:    [Backend]:          Loading Screen Ready
(11:59:07) INFO:    [Frontend]:         Refreshing undefined Library
(11:59:07) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(11:59:07) INFO:    [Legendary]:        Refreshing library...
(11:59:07) INFO:    [Legendary]:        Refreshing Epic Games...
(11:59:07) INFO:    [Legendary]:        Game list updated, got 94 games & DLCs
(11:59:07) INFO:    [Backend]:          Frontend Ready
(11:59:07) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(11:59:07) INFO:    [Gog]:              Found 0 game(s) to update
(11:59:07) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:59:07) INFO:    [Legendary]:        Checking for game updates: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:59:09) INFO:    [Winetricks]:       Downloading Winetricks
(11:59:09) INFO:    [Legendary]:        Found 0 games to update
(11:59:11) INFO:    [WineDownloader]:   Updating wine versions info
(11:59:11) INFO:    [WineDownloader]:   Fetching upstream information...
(11:59:12) INFO:    [Backend]:          Starting the Download Queue
(11:59:13) INFO:    [WineDownloader]:   wine versions updated
(11:59:13) INFO:    [Backend]:          Launching Bloons TD 6 (7786b355a13b47a6b3915335117cd0b2)
(11:59:13) INFO:    [Backend]:          Preventing display from sleep
(11:59:13) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:13) INFO:    [Backend]:          Preventing machine to sleep
(11:59:13) INFO:    [Backend]:          Stopping Power Saver Blocker
(11:59:13) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:13) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:13) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:13) DEBUG:   [Backend]:          Running Wine command: wineboot --init
(11:59:14) INFO:    [DXVKInstaller]:    installing dxvk on... /home/anon/Games/Heroic/Prefixes/default
(11:59:14) INFO:    [DXVKInstaller]:    dxvk already installed!
(11:59:14) INFO:    [DXVKInstaller]:    installing vkd3d on... /home/anon/Games/Heroic/Prefixes/default
(11:59:14) INFO:    [DXVKInstaller]:    vkd3d already installed!
(11:59:14) INFO:    [Legendary]:        Launching Bloons TD 6: LD_LIBRARY_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib HEROIC_APP_NAME=7786b355a13b47a6b3915335117cd0b2 HEROIC_APP_RUNNER=legendary HEROIC_APP_SOURCE=epic DRI_PRIME=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia DOTNET_BUNDLE_EXTRACT_BASE_DIR= DOTNET_ROOT= WINEPREFIX=/home/anon/Games/Heroic/Prefixes/default WINEDLLOVERRIDES=winemenubuilder.exe=d WINE_FULLSCREEN_FSR=0 WINEESYNC=1 WINEFSYNC=1 ORIG_LD_LIBRARY_PATH= GST_PLUGIN_SYSTEM_PATH_1_0=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/gstreamer-1.0:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/gstreamer-1.0 WINEDLLPATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/wine:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/wine LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 7786b355a13b47a6b3915335117cd0b2 --wine /home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/bin/wine --wrapper /app/bin/gamemoderun --language pt_BR
(11:59:19) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(11:59:19) INFO:    [Frontend]:         Refreshing legendary Library
(11:59:19) INFO:    [Legendary]:        Refreshing library...
(11:59:19) INFO:    [Legendary]:        Refreshing Epic Games...
(11:59:19) INFO:    [Legendary]:        Game list updated, got 94 games & DLCs
(11:59:20) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:59:22) INFO:    [Backend]:          Launching The Alto Collection (Hornbill)
(11:59:22) INFO:    [Backend]:          Preventing display from sleep
(11:59:22) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:22) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:22) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:22) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:22) DEBUG:   [Backend]:          Running Wine command: wineboot --init
(11:59:22) INFO:    [Backend]:          Stopping Power Saver Blocker
(11:59:23) INFO:    [DXVKInstaller]:    installing dxvk on... /home/anon/Games/Heroic/Prefixes/default/The Alto Collection
(11:59:23) INFO:    [DXVKInstaller]:    dxvk already installed!
(11:59:23) INFO:    [DXVKInstaller]:    installing vkd3d on... /home/anon/Games/Heroic/Prefixes/default/The Alto Collection
(11:59:23) INFO:    [DXVKInstaller]:    vkd3d already installed!
(11:59:23) INFO:    [Legendary]:        Launching The Alto Collection: LD_LIBRARY_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib HEROIC_APP_NAME=Hornbill HEROIC_APP_RUNNER=legendary HEROIC_APP_SOURCE=epic DRI_PRIME=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia DOTNET_BUNDLE_EXTRACT_BASE_DIR= DOTNET_ROOT= WINEPREFIX="/home/anon/Games/Heroic/Prefixes/default/The Alto Collection" WINEDLLOVERRIDES=winemenubuilder.exe=d WINE_FULLSCREEN_FSR=0 WINEESYNC=1 WINEFSYNC=1 ORIG_LD_LIBRARY_PATH= GST_PLUGIN_SYSTEM_PATH_1_0=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/gstreamer-1.0:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/gstreamer-1.0 WINEDLLPATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/wine:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/wine LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Hornbill --wine /home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/bin/wine --wrapper /app/bin/gamemoderun --language pt_BR
(11:59:27) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(11:59:28) INFO:    [Frontend]:         Refreshing legendary Library
(11:59:28) INFO:    [Legendary]:        Refreshing library...
(11:59:28) INFO:    [Legendary]:        Refreshing Epic Games...
(11:59:28) INFO:    [Legendary]:        Game list updated, got 94 games & DLCs
(11:59:28) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:59:29) INFO:    [Backend]:          Launching Darkwood (923130ebb546417b9d3115507f752d34)
(11:59:29) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:29) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:29) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:29) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-latest
(11:59:29) DEBUG:   [Backend]:          Running Wine command: wineboot --init
(11:59:29) INFO:    [Backend]:          Stopping Power Saver Blocker
(11:59:30) INFO:    [DXVKInstaller]:    installing dxvk on... /home/anon/Games/Heroic/Prefixes/default/Darkwood
(11:59:30) INFO:    [DXVKInstaller]:    dxvk already installed!
(11:59:30) INFO:    [DXVKInstaller]:    installing vkd3d on... /home/anon/Games/Heroic/Prefixes/default/Darkwood
(11:59:30) INFO:    [DXVKInstaller]:    vkd3d already installed!
(11:59:30) INFO:    [Legendary]:        Launching Darkwood: LD_LIBRARY_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib HEROIC_APP_NAME=923130ebb546417b9d3115507f752d34 HEROIC_APP_RUNNER=legendary HEROIC_APP_SOURCE=epic DRI_PRIME=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia DOTNET_BUNDLE_EXTRACT_BASE_DIR= DOTNET_ROOT= WINEPREFIX=/home/anon/Games/Heroic/Prefixes/default/Darkwood WINEDLLOVERRIDES=winemenubuilder.exe=d WINE_FULLSCREEN_FSR=0 WINEESYNC=1 WINEFSYNC=1 ORIG_LD_LIBRARY_PATH= GST_PLUGIN_SYSTEM_PATH_1_0=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/gstreamer-1.0:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/gstreamer-1.0 WINEDLLPATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib64/wine:/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/lib/wine LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 923130ebb546417b9d3115507f752d34 --wine /home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-latest/bin/wine --wrapper /app/bin/gamemoderun --language pt_BR
(11:59:34) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(11:59:35) INFO:    [Frontend]:         Refreshing legendary Library
(11:59:35) INFO:    [Legendary]:        Refreshing library...
(11:59:35) INFO:    [Legendary]:        Refreshing Epic Games...
(11:59:35) INFO:    [Legendary]:        Game list updated, got 94 games & DLCs
(11:59:35) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
```

### Steps to reproduce

1. Make sure you are under a X11 session (tested under KDE Plasma X11)
2. Download any game (or click on settings for any game)
3. Make sure to select "use dedicated graphics card"
4. Game launches, then closes instantly after
5. No errors reported in the logs.

### Expected behavior

Games able to run under X11 when selecting "use dedicated graphics card"

### Heroic Version

Latest Stable (Flatpak)

### System Information

CPU: 8x Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
Memory: 16.65 GB
GPUs:
  GPU 0:
    Name: NVIDIA Corporation GP107M [GeForce GTX 1050 Ti Mobile]
    IDs: D=1c8c V=10de SD=118b SV=1025
    Driver: nvidia
  GPU 1:
    Name: Intel Corporation HD Graphics 630
    IDs: D=591b V=8086 SD=118a SV=1025
    Driver: i915
OS: Arch Linux 6.6.1-arch1-1 (linux)

The current system is not a Steam Deck
We are running inside a Flatpak container

Software Versions:
  Heroic: 2.10.0 Jewelry Bonney
  Legendary: 0.20.33 Undue Alarm
  gogdl: 0.7.3
  Nile: 1.0.0 Jonathan Joestar

### Additional information

If "use dedicated graphics card" is not selected, all games launch normally.

Deselecting "Enable Esync" and "Enable Fsync" did not change the outcome.

**Comment by 0x177** [comment]
did you find a solution? i have the same hardware

**Comment by MrQvest** [comment]
Unfortunately no. In the meantime I use Wayland whenever I want to play heroic games

**Comment by MrQvest** [comment]
Update to version 2.11.0 Kumachi. Still not working, but here's the log from Bloons TD 6:

# Log for Bloons TD 6
```
System Info:
CPU: 8x Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
Memory: 16.65 GB (used: 5.82 GB)
GPUs:
  GPU 0:
    Name: NVIDIA Corporation GP107M [GeForce GTX 1050 Ti Mobile]
    IDs: D=1c8c V=10de SD=118b SV=1025
    Driver: nvidia
  GPU 1:
    Name: Intel Corporation HD Graphics 630
    IDs: D=591b V=8086 SD=118a SV=1025
    Driver: i915
OS: Arch Linux 6.6.3-arch1-1 (linux)

The current system is not a Steam Deck
We are running inside a Flatpak container

Software Versions:
  Heroic: 2.11.0 Kumachi
  Legendary: 0.20.33 Undue Alarm
  gogdl: 0.7.3
  Nile: 1.0.0 Jonathan Joestar

Game Settings: {
	"autoInstallDxvk": true,
	"autoInstallVkd3d": true,
	"preferSystemLibs": false,
	"enableEsync": true,
	"enableFsync": true,
	"nvidiaPrime": true,
	"enviromentOptions": [],
	"wrapperOptions": [],
	"showFps": false,
	"useGameMode": true,
	"language": "",
	"wineVersion": {
		"bin": "/home/anon/.local/share/Steam/steamapps/common/Proton - Experimental/proton",
		"name": "Proton - Proton - Experimental",
		"type": "proton"
	},
	"winePrefix": "/home/anon/Games/Heroic/Prefixes/default/Bloons TD 6",
	"wineCrossoverBottle": ""
}

Launch Command: HEROIC_APP_NAME=7786b355a13b47a6b3915335117cd0b2 HEROIC_APP_RUNNER=legendary HEROIC_APP_SOURCE=epic DRI_PRIME=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia DOTNET_BUNDLE_EXTRACT_BASE_DIR= DOTNET_ROOT= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/anon/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH="/home/anon/Games/Heroic/Prefixes/default/Bloons TD 6" STEAM_COMPAT_INSTALL_PATH=/home/anon/Games/Heroic/BloonsTD6 WINE_FULLSCREEN_FSR=0 STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=heroic-BloonsTD6 PROTON_LOG_DIR=/home/anon/.var/app/com.heroicgameslauncher.hgl /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 7786b355a13b47a6b3915335117cd0b2 --no-wine --wrapper "/app/bin/gamemoderun '/home/anon/.local/share/Steam/steamapps/common/Proton - Experimental/proton' run" --language pt_BR

Game Log:
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
gamemodeauto: 
gamemodeauto: 
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API FAIL] SteamAPI_Init() failed
[UnityMemory] Configuration Parameters - Can be set up in boot.config
    "memorysetup-bucket-allocator-granularity=16"
    "memorysetup-bucket-allocator-bucket-count=8"
    "memorysetup-bucket-allocator-block-size=4194304"
    "memorysetup-bucket-allocator-block-count=1"
    "memorysetup-main-allocator-block-size=16777216"
    "memorysetup-thread-allocator-block-size=16777216"
    "memorysetup-gfx-main-allocator-block-size=16777216"
    "memorysetup-gfx-thread-allocator-block-size=16777216"
    "memorysetup-cache-allocator-block-size=4194304"
    "memorysetup-typetree-allocator-block-size=2097152"
    "memorysetup-profiler-bucket-allocator-granularity=16"
    "memorysetup-profiler-bucket-allocator-bucket-count=8"
    "memorysetup-profiler-bucket-allocator-block-size=4194304"
    "memorysetup-profiler-bucket-allocator-block-count=1"
    "memorysetup-profiler-allocator-block-size=16777216"
    "memorysetup-profiler-editor-allocator-block-size=1048576"
    "memorysetup-temp-allocator-size-main=4194304"
    "memorysetup-job-temp-allocator-block-size=2097152"
    "memorysetup-job-temp-allocator-block-size-background=1048576"
    "memorysetup-job-temp-allocator-reduction-small-platforms=262144"
    "memorysetup-allocator-temp-initial-block-size-main=262144"
    "memorysetup-allocator-temp-initial-block-size-worker=262144"
    "memorysetup-temp-allocator-size-background-worker=32768"
    "memorysetup-temp-allocator-size-job-worker=262144"
    "memorysetup-temp-allocator-size-preload-manager=262144"
    "memorysetup-temp-allocator-size-nav-mesh-worker=65536"
    "memorysetup-temp-allocator-size-audio-worker=65536"
    "memorysetup-temp-allocator-size-cloud-worker=32768"
    "memorysetup-temp-allocator-size-gfx=262144"
Initialize engine version: 2022.3.7f1 (b16b3b16c7a0)
[Subsystems] Discovering subsystems at path Z:/home/anon/Games/Heroic/BloonsTD6/BloonsTD6_Data/UnitySubsystems
GfxDevice: creating device client; threaded=1; jobified=0
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1050 Ti (ID=0x73df)
    Vendor:   ATI
    VRAM:     4342 MB
    Driver:   31.0.21902.5
<RI> Initializing input.

New input system (experimental) initialized
D3D11 device created for Microsoft Media Foundation video decoding.
Using Windows.Gaming.Input
<RI> Input initialized.

X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  156 (NV-GLX)
  Minor opcode of failed request:  43 ()
  Serial number of failed request:  409
  Current serial number in output stream:  410
gamemodeauto: 
```

**Comment by soumyaDghosh** [comment]
Can confirm this bug. Faced with Cat Quest

**Comment by kokoko3k** [comment]
This happens when using proton(ish ) 8.0+

There are several reports online; what i discovered is that the problem lies in the __NV_PRIME_RENDER_OFFLOAD=1 environment variable.
Stripping it makes vulkan games work, but shifts the problem to opengl games.

I opened a bug report to nvidia:
https://forums.developer.nvidia.com/t/proton-8-prime-render-offload-wont-work-anymore-under-x11/279987

...and I was about to open a bug report versus proton, but curiously, prime-run script under steam DO work even with proton 8.0, so i'm writing it here now.

Long story short, you should **DISABLE** the option to use the discrete video graphic card in heroic and put an environment variable to force the use of nvidia vulkan drivers:

either:
__VK_LAYER_NV_optimus=NVIDIA_only
or something like:
VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json (depends on where nvidia_icd.json is)

...I'm not sure what is the best way to handle this by Heroic side.

**Comment by Hmegh** [comment]
I had the same issue, switching to Proton Experimental fixed it. Did you try it?
